src/vos/vos_tree.c

```diff
@@ -1354,3 +1355,13 @@ obj_tree_find_attr(unsigned tree_class, int flags)
 			return NULL;
 	}
 }
+
+bool
+vos_irec_is_valid(const struct vos_irec_df *svt, uint32_t dtx_lid)
+{
+	if (svt == NULL) {
```

Comment:
Would need to check but if we use zalloc, we may also be able to check that ir_pad8 and ir_pad16 are zero.  Could possibly hijack one of them  to check if zero || magic

Reply:
According to my quick survey, it is zalloc'ed. However, I am a bit worried about using a side effect as a basis for a new validation rule for this type of record. So, I would not go this way.

Reply:
Sure, my assertion is that we could take one of those fields, rename it to "magic", set a magic number and then allow 0 or magic.  We are essentially preventing corruption here so I think using whatever we can for validation is reasonable.  @NiuYawei @Nasf-Fan have an opinion here?

Reply:
Along similar lines, I wonder if there is anything else we can use to validate the record(s).   Such as whether the bio_addr value makes sense (e.g. not out of bounds).

Reply:
I just filed a ticket to refine this idea and possibly implement it in the next round: https://daosio.atlassian.net/browse/DAOS-17005. Feel free to add further ideas there. I believe there is a greater chance to discuss them properly and make an informed decision.